strawberryfields/configuration.py

```diff
@@ -15,138 +15,229 @@
+        port (int): the port to be used when connecting to the remote service
+        debug (bool): determines if the debugging mode is requested
+
+    Returns:
+        dict[str, dict[str, Union[str, bool, int]]]: the configuration
+            object
+    """
+    hostname = kwargs.get("hostname", "localhost")
+    use_ssl = kwargs.get("use_ssl", True)
+    port = kwargs.get("port", 443)
+    debug = kwargs.get("debug", False)
+
+    config = {
+        "api": {
+            "authentication_token": authentication_token,
+            "hostname": hostname,
+            "use_ssl": use_ssl,
+            "port": port,
+            "debug": debug
+            }
+    }
+    return config
+
+def look_for_config_in_file(filename="config.toml"):
+    """Looks for the first configuration file to be found at certain paths.
+
+    .. note::
+
+        The following directories are checked (in the following order):
+
+        * The current working directory
+        * The directory specified by the environment variable SF_CONF (if specified)
+        * The user configuration directory (if specified)
+
+    Keyword arguments:
+        filename (str): the configuration file to look for
+
+    Returns:
+         dict[str, dict[str, Union[str, bool, int]]] or None: the
+             configuration object that was loaded
+    """
+
+    # Search the current directory, the directory under environment
+    # variable SF_CONF, and default user config directory, in that order.
+    current_dir = os.getcwd()
+    sf_env_config_dir = os.environ.get("SF_CONF", "")
+    sf_user_config_dir = user_config_dir("strawberryfields", "Xanadu")
+
+    directories = [current_dir, sf_env_config_dir, sf_user_config_dir]
+    for directory in directories:
+        filepath = os.path.join(directory, filename)
+        try:
+            parsed_config = load_config_file(filepath)
+            break
+        except FileNotFoundError:
+            parsed_config, filepath = None, None
+
+    return parsed_config, filepath
+
+def load_config_file(filepath):
+    """Load a configuration object from a TOML formatted file.
+
+    Args:
+        filepath (str): path to the configuration file
+
+    Returns:
+         dict[str, dict[str, Union[str, bool, int]]]: the configuration
+            object that was loaded
+    """
+    with open(filepath, "r") as f:
+        config_from_file = toml.load(f)
+    return config_from_file
+
+def update_with_other_config(config, other_config):
+    """Updates the current configuration object with another one.
+
+    Args:
+        config (dict[str, dict[str, Union[str, bool, int]]]): the
+            configuration to be updated
+        other_config (dict[str, dict[str, Union[str, bool, int]]]): the
+            configuration used for updating
+
+    Returns:
+        dict[str, dict[str, Union[str, bool, int]]]): the updated
+            configuration
+    """
+    # Here an example for sectionconfig is API
```

Comment:
Yes, so this is connected to the repeated use of the word "Currently" :smile: The comment just wanted to indicate that the `"api"` is an example for a `sectionconfig` (and actually the only example for now). However, since it seems to be confusing, I'll just remove it :)

Reply:
I just realized that the entire confusion comes from the fact that the comment itself should say `section` instead of `sectionconfig`, as each `section` is a key in the outer dictionary. Would it be a useful comment as such or would it be better to remove it as intended?

Reply:
new version looks fine :+1: